File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/344. Reverse String/C++ solution/344. Reverse String.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        int L = 0, R = s.size() - 1;
        while (L < R)
        {
            char temp = s[L];
            s[L] = s[R];
            s[R] = temp;
            L++;
            R--;
        }

        // using the reverse method
        // reverse(s.begin(), s.end());
    }
};

File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/344. Reverse String/Python solution/344. Reverse String.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        L, R = 0, len(s) - 1
        while L < R:
            s[L], s[R] = s[R], s[L]
            L += 1
            R -= 1
        
        # OR simply use the revers method
        # s.reverse()